src/storage/src/hummock/event_handler/uploader/mod.rs

```diff
@@ -807,7 +826,20 @@ impl UnsyncData {
             .get_mut(&instance_id)
             .expect("should exist");
         let epoch = instance_data.local_seal_epoch(next_epoch);
-        self.epochs.insert(next_epoch, ());
+        // When drop/cancel a streaming job, for the barrier to stop actor, the
+        // local instance will call `local_seal_epoch`, but the `next_epoch` won't be
+        // called `start_epoch` because we have stopped writing on it.
+        if !table_data.unsync_epochs.contains_key(&next_epoch) {
+            if let Some(stopped_next_epoch) = table_data.stopped_next_epoch {
+                assert_eq!(stopped_next_epoch, next_epoch);
+            } else {
+                if let Some(max_epoch) = table_data.max_epoch() {
+                    assert_gt!(next_epoch, max_epoch);
+                }
+                debug!(?table_id, epoch, next_epoch, "table data has stopped");
+                table_data.stopped_next_epoch = Some(next_epoch);
```

Comment:
so `stopped_next_epoch` is introduced only for assertion check?

Reply:
Yes.